flink-table-store-spark/src/main/java/org/apache/flink/table/store/spark/SparkTable.java

```diff
@@ -76,4 +94,47 @@ public Transform[] partitioning() {
         }
         return new Transform[0];
     }
+
+    @Override
+    public WriteBuilder newWriteBuilder(LogicalWriteInfo info) {
+        try {
+            // V1Write is introduced in Spark 3.2
+            // Use reflection to support Spark 3.0 and 3.1
+            return (WriteBuilder)
+                    Class.forName("org.apache.flink.table.store.spark.SparkWriteBuilder")
+                            .getConstructors()[0]
+                            .newInstance(table, info.queryId(), lockFactory);
+        } catch (ClassNotFoundException
+                | IllegalAccessException
+                | InvocationTargetException
+                | InstantiationException e) {
+            throw new UnsupportedOperationException(
+                    "Please use Spark version 3.2 or above to support write.", e);
```

Comment:
```suggestion
                    "Please use Spark version 3.0 or above to support write.", e);
```

Reply:
It is spark 3.2.
3.0 and 3.1 do not support.